packager/hls/base/master_playlist_unittest.cc

```diff
@@ -138,7 +139,8 @@ class MasterPlaylistTest : public ::testing::Test {
   MasterPlaylistTest()
       : master_playlist_(kDefaultMasterPlaylistName,
                          kDefaultAudioLanguage,
-                         kDefaultTextLanguage),
+                         kDefaultTextLanguage,
+                         kSegmentSapAligned),
```

Comment:
Let's keep it as false here, i.e. "!kSegmentSapAligned" (btw, the variable should be renamed).

Can you also make |master_playlist_| a unique_ptr?

```c++
std::unique_ptr<MasterPlaylist> master_playlist_;
```

The code can be changed to:

```c++
master_playlist_(new MasterPlaylist(...));
```

Then add a new unit-test similar to WriteMasterPlaylistOneVideo, e.g. WriteMasterPlaylistOneVideoWithIndependentSegments.

Reply:
what should kSegmentSapAligned be renamed to? kIsIndependentSegments?

Reply:
Also, how should the additional unit test with independent segments be written? Currently, independent segments is initialized to false, but there isn't some other constructor that would initialize it to true, which is needed to test the playlist with independent segments.